Extensions/Physics3DBehavior/JsExtension.js

```diff
@@ -0,0 +1,2463 @@
+              .getChild('sidewaysSpeedMax')
+              .getDoubleValue()
+              .toString(10)
+          )
+          .setAdvanced(true);
+
+        behaviorProperties
+          .getOrCreate('slopeMaxAngle')
+          .setLabel('Slope max. angle')
+          .setGroup(_('Walk'))
+          .setType('Number')
+          .setMeasurementUnit(gd.MeasurementUnit.getDegreeAngle())
+          .setValue(
+            behaviorContent
+              .getChild('slopeMaxAngle')
+              .getDoubleValue()
+              .toString(10)
+          )
+          .setAdvanced(true)
+          .setQuickCustomizationVisibility(gd.QuickCustomization.Hidden);
+
+        behaviorProperties
+          .getOrCreate('shouldBindObjectAndForwardAngle')
+          .setLabel('Keep object angle and forward direction the same')
+          .setGroup(_('Walk'))
+          .setType('Boolean')
+          .setValue(
+            behaviorContent
+              .getChild('shouldBindObjectAndForwardAngle')
+              .getBoolValue()
+              ? 'true'
+              : 'false'
+          )
+          .setAdvanced(true)
+          .setQuickCustomizationVisibility(gd.QuickCustomization.Hidden);
+
+        return behaviorProperties;
+      };
+
+      behavior.initializeContent = function (behaviorContent) {
+        behaviorContent.addChild('physics3D').setStringValue('');
+        behaviorContent.addChild('jumpHeight').setDoubleValue(200);
+        behaviorContent.addChild('jumpSustainTime').setDoubleValue(0.2);
+        behaviorContent.addChild('gravity').setDoubleValue(1000);
+        behaviorContent.addChild('fallingSpeedMax').setDoubleValue(700);
+        behaviorContent.addChild('forwardAcceleration').setDoubleValue(1200);
+        behaviorContent.addChild('forwardDeceleration').setDoubleValue(1200);
+        behaviorContent.addChild('forwardSpeedMax').setDoubleValue(600);
+        behaviorContent.addChild('sidewaysAcceleration').setDoubleValue(800);
+        behaviorContent.addChild('sidewaysDeceleration').setDoubleValue(800);
+        behaviorContent.addChild('sidewaysSpeedMax').setDoubleValue(400);
+        behaviorContent.addChild('slopeMaxAngle').setDoubleValue(50);
+        behaviorContent
+          .addChild('shouldBindObjectAndForwardAngle')
+          .setBoolValue(true);
+      };
+
+      const aut = extension
+        .addBehavior(
+          'PhysicsCharacter3D',
+          _('3D physics character'),
+          'PhysicsCharacter3D',
+          _('Jump and run on platforms.'),
+          '',
+          'JsPlatform/Extensions/physics_character3d.svg',
+          'PhysicsCharacter3D',
+          //@ts-ignore The class hierarchy is incorrect leading to a type error, but this is valid.
+          behavior,
+          new gd.BehaviorsSharedData()
+        )
+        .addIncludeFile(
+          'Extensions/Physics3DBehavior/PhysicsCharacter3DRuntimeBehavior.js'
+        );
+
+      aut
+        .addScopedAction(
+          'SimulateForwardKey',
```

Comment:
Open question: I wonder if we should have a single "SimulateKey" that takes a key as a parameter.

Reply:
Probably at some point, but we can't remove or rename a choice from `stringWithSelector` without creating a new instruction so I prefer not to do it until we are sure about the controls.